eclipse-collections-api/src/main/java/org/eclipse/collections/api/RichIterable.java

```diff
@@ -295,17 +295,38 @@ else if (inside.size() > 32 && !(inside instanceof SetIterable))
     /**
      * Returns true if all elements in source are contained in this collection.
      *
-     * @since 1.0
+     * @since 11.1
      */
-    boolean containsAllIterable(Iterable<?> source);
+    default boolean containsAllIterable(Iterable<?> source)
+    {
+        if (source instanceof RichIterable)
+        {
+            RichIterable<?> outside = this;
+            RichIterable<?> inside = (RichIterable<?>) source;
+            if (this.size() < inside.size())
+            {
+                return false;
+            }
```

Comment:
On line 306, we've already called size() on both collections. If the sizes are unequal, we can return false. In other words, replace `<` with `!=`.

Reply:
Would this be correct? `A.containsAll(B)` should return `true` iff all the elements of `B` are contained in `A`, so `A` and `B` don't necessarily have the same size.

Reply:
If A and B don't have the same size then they are definitely not equal.